File: 381A Sereja and Dima/SerejaAndDima.java
import java.util.*;

public class SerejaAndDima {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            int n = sc.nextInt();
            Deque<Integer> d = new LinkedList<>();
            for (int i = 0; i < n; i++)
                d.add(sc.nextInt());
            int sc1 = 0, sc2 = 0;
            int i = 0;
            while (!d.isEmpty()) {
                int l;
                if (d.peekLast() > d.peekFirst())
                    l = d.pollLast();
                else
                    l = d.pollFirst();
                if (i++ % 2 == 0)
                    sc1 += l;
                else
                    sc2 += l;
            }
            System.out.println(sc1 + " " + sc2);
        }
    }
}

File: 381A Sereja and Dima/SerejaAndDima.py


from collections import deque


def main():
    n = int(input())
    d = deque()
    [d.append(int(x)) for x in input().split()]
    sc1 = 0
    sc2 = 0
    i = 0
    while len(d) > 0:
        if d[0] > d[-1]:
            l = d[0]
            d.popleft()
        else:
            l = d[-1]
            d.pop()
        if i % 2 == 0:
            sc1 += l
        else:
            sc2 += l
        i += 1
    print(sc1, sc2)


if __name__ == "__main__":
    main()
